apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/pagerules/PageRuleFinder.java

```diff
@@ -0,0 +1,140 @@
+           [rule].logic                       as [comparator],
+           [rule].target_type                 as [targetType],
+           [rule].target_question_identifier  as [targetQuestions],
+           0                                  as [TotalCount]
+       from WA_rule_metadata [rule]
+       where   [rule].wa_rule_metadata_uid =:ruleId
+       """;
+
+  private  String findByPageId = """   
+      select
+           [rule].wa_rule_metadata_uid        as [ruleId],
+           [rule].wa_template_uid             as [template],
+           [rule].rule_cd                     as [function],
+           [rule].rule_desc_txt               as [description],
+           [rule].source_question_identifier  as [sourceQuestion],
+           [rule].rule_expression             as [ruleExpression],
+           [rule].source_values               as [sourceValues],
+           [rule].logic                       as [comparator],
+           [rule].target_type                 as [targetType],
+           [rule].target_question_identifier  as [targetQuestions],
+           (SELECT COUNT(*) 
+                  FROM WA_rule_metadata [rule] 
+                      WHERE [rule].wa_rule_metadata_uid = :pageId)
+                                              as [TotalCount]
+       from WA_rule_metadata [rule]
+       where   [rule].wa_template_uid =:pageId
+       order by add_time
+       offset :offset rows
+       fetch next :pageSize rows only
+       """;
+
+  private  String findBySearchValue = """   
+      select
+           [rule].wa_rule_metadata_uid        as [ruleId],
+           [rule].wa_template_uid             as [template],
+           [rule].rule_cd                     as [function],
+           [rule].rule_desc_txt               as [description],
+           [rule].source_question_identifier  as [sourceQuestion],
+           [rule].rule_expression             as [ruleExpression],
+           [rule].source_values               as [sourceValues],
+           [rule].logic                       as [comparator],
+           [rule].target_type                 as [targetType],
+           [rule].target_question_identifier  as [targetQuestions],
+           (SELECT COUNT(*) 
+                  FROM WA_rule_metadata [rule]  
+                     where    UPPER(source_values) LIKE CONCAT('%', UPPER(:searchValue), '%') OR 
+                     UPPER(target_question_identifier) LIKE CONCAT('%', UPPER(:searchValue), '%')
+           ) as [TotalCount] 
+                                                
+       from WA_rule_metadata [rule]
+       where    UPPER(source_values) LIKE CONCAT('%', UPPER(:searchValue), '%') OR 
+                UPPER(target_question_identifier) LIKE CONCAT('%', UPPER(:searchValue), '%')
+       order by add_time
+       offset :offset rows
+       fetch next :pageSize rows only
+       """;
+
+  private final NamedParameterJdbcTemplate template;
+  private final RowMapper<RuleResponse> mapper;
```

Comment:
Question: Can we not map directly to a `Rule` instead of `RuleResponse` and then mapping it again to a `Rule` in the `PageRuleReader`?